### PR TITLE
Fixing in treating empty lists as a conjunction.

### PR DESCRIPTION
**Pull request by MrThreepwood**
# Motivation:

This should prevent the unexpected behavior when providing empty lists to the in predicate function. Currently doing something like 
```kotlin
selectAll()
from<Contract>()
whereAnd(column(Contract::userId).`in`(emptyList()))
```
returns every record on the table. This has the potential to cause data leaks (and would currently cause this exact leak in our code if we had pushed it to production).

# Modifications:

Changed InValueSpec to treat empty lists as a disjunction instead of a conjunction.

# Result:

Closes issue #191 

Queries will no longer ignore predicates containing empty lists, treating them as always false. I did not see any mention of its current behavior in the documentation, so I expect this change will likely result in expected behavior for most users as the current behavior is fairly surprising.

The current behavior has the potential for catastrophic data breaches and loss of data. I believe that the appropriate response here would be to report a critical vulnerability with a CVE organization (such as Snyk) regardless of whether this PR is merged in. I don't expect most users of this library are aware of this behavior, and had I not found it myself I would have greatly appreciated an alert from our dependency security scans.

### CODE REVIEW

**Comment by cj848** [comment]
The current test is failing. Fix the failing test. And when the disjunction method is reflected and an empty list is entered, the actual query is executed and the data is not deleted. It would be nice to check the tests in the classes ending with IntegrationTest .

**Comment by cj848** [comment]
Test passed. Thank you for your effort. However, the commit message was too long to pass lint. This part is going to be included as a breaking change in the release note, so I think it can be removed. And please agree to the CLA.

**Comment by MrThreepwood** [comment]
I shortened the commit message, and I agreed to the license, but it doesn't seem to work. It seems to not like the user in my commit (Joshua Swoyer, the same name as I used to sign it).

I'm not concerned about credit, so if it's easier for you to reset my commit and push it up yourself I have no issues with that. Obviously the changes I made are so simple it would be impossible to claim copyright anyway. Otherwise if you have any documentation on how to get the license to respect my signature I can try that.

**Review by cj848 — APPROVED** [review]

**Comment by cj848** [comment]
CLA works normally only when git configuration is included.

Even if it's not a big problem right now, it would be nice to set up git for the next contribution.

@shown, @pickmoment @huisam Please review and I will merge and release as soon as it is confirmed.

**Review by shouwn — APPROVED** [review]
I'm sorry I checked too late.... Thank you for your contribution!

**Review by huisam — APPROVED** [review]

**Comment by cj848** [comment]
@pickmoment I think you're very busy, so I'll wait until today and if there's no separate approval, I'll release it because there was approval from the rest of the people.

**Pull request merged**
